src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs

```diff
@@ -132,17 +130,12 @@ public override bool Execute()
                 log.LogDebug($"(in) RestoreConfigFile '{RestoreConfigFile}'");
             }
 
-            if (RestoreDirectory != null)
-            {
-                log.LogDebug($"(in) RestoreDirectory '{RestoreDirectory}'");
-            }
-
             if (RestoreSolutionDirectory != null)
             {
                 log.LogDebug($"(in) RestoreSolutionDirectory '{RestoreSolutionDirectory}'");
             }
 
-            var settings = ReadSettings(RestoreSolutionDirectory, string.IsNullOrEmpty(RestoreDirectory) ? Path.GetDirectoryName(ProjectUniqueName) : RestoreDirectory, RestoreConfigFile);
+            var settings = ReadSettings(RestoreSolutionDirectory, Path.GetDirectoryName(ProjectUniqueName) , RestoreConfigFile);
```

Comment:
Will do it later. 
Updated the Issue with this comment:
https://github.com/NuGet/Home/issues/5181